src/pcost.py

```diff
@@ -11,12 +12,16 @@ def portfolio_cost(file_path: Union[str, Path]) -> float:
     :param file_path: Path to the portfolio file
     :return: The cost of the portfolio
     """
-    portfolio = parse_csv(
+    portdict = parse_csv(
         file_path,
         select=["shares", "price"],
         convert_fn={"shares": int, "price": float},
     )
-    portfolio = cast(List[Dict[str, Any]], portfolio)
-    total_cost = sum([p["shares"] * p["price"] for p in portfolio])
+    portfolio = [
+        Stock(name="", shares=pd["shares"], price=pd["price"])
+        for pd in portdict
+        if isinstance(pd, dict)
+    ]
+    total_cost = sum([p.cost for p in portfolio])
```

Comment:
You can omit the list creation here and use directly:
```suggestion
    total_cost = sum(p.cost for p in portfolio)
```

On of the main difference is that in your implementation it:
1. creates the list thanks to the list comprehension
2. applies the `sum` on the list created

In the implementation that I suggested it uses a generator expression so you do not create the full list as a first step. One of the main advantage of using generator expression is the memory used: if the list on which the sum is applied is really large you can have a pick of memory consumption. Using generator allows to avoid this kind of issue and there is no impact on the time to compute this sum!

Reply:
Fixed